internal/terraform/terraform.go

```diff
@@ -73,21 +74,63 @@ func GetResourceType(resource hclwrite.Block) string {
 	return resource.Labels()[0]
 }
 
-func ValidateTerraformInitRun(dir string) error {
-	_, err := os.Stat(dir + "/.terraform")
+func ValidateInitRun(dir string, terragrunt bool) error {
+	path := dir + "/."
+	if terragrunt {
+		path += "terragrunt-cache"
+	} else {
+		path += "terraform"
+	}
+
+	if _, err := os.Stat(path); err != nil {
+		var errorTypeStr string
+		if terragrunt {
+			errorTypeStr = "terragrunt"
+		} else {
+			errorTypeStr = "terraform"
+		}
 
-	if err != nil {
 		if os.IsNotExist(err) {
-			return errors.New("terraform init must run before running terratag")
+			return fmt.Errorf("%s init must run before running terratag", errorTypeStr)
 		}
 
-		return fmt.Errorf("couldn't determine if terraform init has run: %v", err)
+		return fmt.Errorf("couldn't determine if %s init has run: %v", errorTypeStr, err)
 	}
 
 	return nil
 }
 
-func GetTerraformFilePaths(rootDir string) ([]string, error) {
+func GetFilePaths(dir string, terragrunt bool) ([]string, error) {
+	if terragrunt {
+		return getTerragruntFilePath(dir)
+	} else {
+		return getTerraformFilePaths(dir)
+	}
+}
+
+func getTerragruntFilePath(rootDir string) ([]string, error) {
+	rootDir += "/.terragrunt-cache"
```

Comment:
Can we DRY this constant a bit? i.e using some common func of `getRootDir` that receives an IaC type

Reply:
Yes!

Reply:
refactored.